windows_bat/PokemonGo-Bot-Install.bat

```diff
@@ -44,45 +44,79 @@ exit /B
 
 
 :gotPrivileges
-setlocal & pushd .
+SETlocal & pushd .
 cd /d %~dp0
 if '%1'=='ELEV' (del "%vbsGetPrivileges%" 1>nul 2>nul & shift /1)
 
 
 
 :detectOS
-reg Query "HKLM\Hardware\Description\System\CentralProcessor\0" | find /i "x86" > NUL && set OS=32-BIT || set OS=64-BIT
+reg Query "HKLM\Hardware\Description\System\CentralProcessor\0" | find /i "x86" > NUL && SET OS=32-BIT || SET OS=64-BIT
+
+ECHO.----- Checking OS Architecture ----->%log%
+ECHO.>>%log%
+ECHO.The computer has "%OS%" architecture.>>%log%
+ECHO.>>%log%
 
 
 
 :CheckInstallPath
-CLS
-@ECHO --------------------Installation Path--------------------
-@ECHO.
-@ECHO.
-@ECHO.
-set InstallPath=
-set /p InstallPath= "Choose an installation folder or press Enter to close:" ||goto:eof
-FOR /F "tokens=1-4 delims=/-. " %%G IN ("%InstallPath%") DO (set InstallPath=%%G\%%H\%%I\%%J)
-set PGBotPath=%InstallPath%\PokemonGo-Bot
-set DownPath=%InstallPath%\Install-Files
-if not exist %DownPath% md %DownPath%
-if "%~dp0"=="%PGBotPath%\windows_bat\" (
+cls
+@ECHO.--------------------Installation Path--------------------
+@ECHO.
+@ECHO.
+@ECHO.
+SET InstallPath=
+SET /p InstallPath= "Choose an installation folder (example: C:/Test/) or press Enter to close: " ||goto:eof
+ECHO.----- Input Path ----->>%log%
+ECHO.>>%log%
+ECHO.The input Path is "%InstallPath%">>%log%
+ECHO.>>%log%
+set InstallPath=%InstallPath:/=\%
+ECHO.----- Converted Path ----->>%log%
+ECHO.>>%log%
+ECHO.The converted Path is "%InstallPath%">>%log%
+ECHO.>>%log%
+SET PGBotPath=%InstallPath%PokemonGo-Bot
+SET DownPath=%InstallPath%Install-Files
+ECHO.----- Checking Download Path ----->>%log%
+ECHO.>>%log%
+IF NOT EXIST %DownPath% (
+ECHO.No download folder "%DownPath%"
+) else (
+ECHO.%DownPath% exists
+)>>%log%
+ECHO.>>%log%
+IF NOT EXIST %DownPath% md %DownPath%
+ECHO.----- Rechecking Download Path ----->>%log%
+ECHO.>>%log%
+IF NOT EXIST %DownPath% (
+ECHO.No download folder "%DownPath%"
+) else (
+ECHO.%DownPath% exists
+)>>%log%
+ECHO.>>%log%
+if %~dp0==%PGBotPath%\windows_bat\ (
 COPY "%PGBotPath%\windows_bat\PokemonGo-Bot-Install.bat" %InstallPath%
-CALL "%InstallPath%\PokemonGo-Bot-Install.bat"
+CALL "%InstallPath%PokemonGo-Bot-Install.bat"
 exit.
 ) ELSE (
-@ECHO Installation Path OK! Proceeding!
+@ECHO.Installation Path OK^^! Proceeding^^!

```

Comment:
echo can't print out the ! . This is a comment from the microsoft website "To display a pipe (|) or redirection character (< or >) when you are using echo, use a caret character immediately before the pipe or redirection character (for example, ^>, ^<, or ^| ). If you need to use the caret character (^), type two (^^). " For some reasons i need 2 ^ the print out the !.